apps/calendar/test/marionette/lib/views/week.js

```diff
@@ -77,6 +83,153 @@ Week.prototype = {
+      return bottomHourOffset - bottomScroll;
+    }
+    return 0;
+  },
+
+  /**
+   * This method waits until we don't scroll.
+   *
+   * Note this doesn't seem to work properly when the view is just displayed.
+   * That's why waitForHourScrollEnd is still useful.
+   */
+  waitForNoScroll: function() {
+    var prevScrollTop = null;
+    this.client.waitFor(function() {
+      var currentScrollTop = this.scrollTop;
+      if (prevScrollTop !== null && prevScrollTop === currentScrollTop) {
+        return true;
+      }
+      prevScrollTop = currentScrollTop;
+      return false;
+    }.bind(this));
+  },
+
+  /**
+   * This method scrolls slowly to the right hour.
+   *
+   * @param {Number} hour The hour to scroll to.
+   */
+  scrollToHour: function(hour) {
+    this.waitForNoScroll();
+    if (this._isHourDisplayed(hour)) {
+      return;
+    }
+
+    this.client.waitFor(function() {
+      var where = this._cmpHourDisplayed(hour);
+      if (where === 0) {
+        return true;
+      }
+
+      var moveDelta = 70;
+
+      // element is hidden above the visible part
+      // we need to move it up
+      if (where < 0) {
+        this.actions.flick(
+          this.main,
+          100, 0, 100, moveDelta
+        ).perform();
+      }
+      // element is hidden below the visible part.
+      // we need to move it down
+      if (where > 0) {
+        this.actions.flick(
+          this.main,
+          100, moveDelta, 100, 0
+        ).perform();
+      }
+
+      this.waitForNoScroll();
+      return false;
+    }.bind(this));
+  },
+
+  /**
+   * Tap the right square. This might throw if the square is not visible, so you
+   * should use scrollToHour before calling this method.
+   *
+   * @param {Object} options
+   * @param {Number} options.day The day to tap. This needs to be a visible day.
+   * @param {Number} options.hour The hour to tap. It needs to be visible as
+   * well.
+   * @param {Number} [options.left] If present, it allows to tap at a specific
+   * location in the square. It can be negative to tap at the left of the
+   * button. Only one of `left`, `right` can be specified.
+   * @param {Number} [options.right] If present, it allows to tap at a specific
+   * location in the square. It can be negative to tap at the right of the
+   * button.
+   * @param {Number} [options.top] If present, it allows to tap at a specific
+   * location in the square. It can be negative to tap at the top of the
+   * button. Only one of `top`, `bottom` can be spcified.
+   * @param {Number} [options.bottom] If present, it allows to tap at a specific
+   * location in the square. It can be negative to tap at the bottom of the
+   * button.
+   */
+  tapDayHour: function(options) {
+    var day = options.day;
+    var hour = options.hour;
+    var left = options.left;
+    var right = options.right;
+    var bottom = options.bottom;
+    var top = options.top;
+
+    if (left === undefined && right === undefined) {
+      left = 25;
```

Comment:
We can make these values more dynamic, just to be safe, if one day the form factor changes.
```js
var dayElt = this.days[day];
var dayEltWidth = dayElt.rect().width
var dayEltHeight = dayElt.rect().height

if (left === undefined && right === undefined) {
    left = dayEltWidth / 2;
}
if (left === undefined) {
    left = dayEltWidth - right;
}

// and so on
```
How do you feel about it?